src/main/resources/services/graphQL/generateQueryJournalsField.es

```diff
@@ -0,0 +1,154 @@
+//import {toStr} from '@enonic/js-utils';
+
+import {
+	GraphQLInt,
+	GraphQLString,
+	list,
+	nonNull
+} from '/lib/graphql';
+import {query} from '/lib/explorer/journal/query';
+
+
+export function generateQueryJournalsField({
+	glue
+}) {
+	const JOURNAL_OBJECT_TYPE = glue.addObjectType({
+		name: 'Journal',
+		//description:
+		fields: {
+			_id: { type: glue.getScalarType('_id') },
+			_name: { type: glue.getScalarType('_name') },
+			_nodeType: { type: GraphQLString }, // TODO nonNull?
+			_path: { type: glue.getScalarType('_path') },
+			displayName: { type: nonNull(GraphQLString) },
+			endTime: { type: nonNull(GraphQLString) },
+			errorCount: { type: nonNull(GraphQLInt) },
+			errors: { type: list(glue.addObjectType({
+				name: 'JournalErrors',
+				//description:
+				fields: {
+					message: { type: GraphQLString },
+					uri: { type: GraphQLString }
+				}
+			}))},
+			duration: { type: nonNull(GraphQLInt) },
+			name: { type: nonNull(GraphQLString) },
+			startTime: { type: nonNull(GraphQLString) },
+			successCount: { type: nonNull(GraphQLInt) },
+			successes: { type: list(glue.addObjectType({
+				name: 'JournalSuccesses',
+				//description:
+				fields: {
+					message: { type: GraphQLString },
+					uri: { type: GraphQLString }
+				}
+			}))}//,
+			//type: { type: nonNull(GraphQLString) }
+		}
+	}); // JOURNAL_OBJECT_TYPE
+	return {
+		args: {
+			//aggregations
+			count: GraphQLInt,
+			//filters
+			//query
+			//page: GraphQLInt,
+			//perPage: GraphQLInt,
+			sort: GraphQLString,
+			start: GraphQLInt
+		},
+		resolve: (env) => {
+			//log.info(`env:${toStr(env)}`);
+			const journalsRes = query(env.args);
+			//log.info(`journalsRes:${toStr(journalsRes)}`);
+			journalsRes.hits = journalsRes.hits.map(({
```

Comment:
What it's actually doing is removing properties...

Reply:
Not exposing _indexConfig etc, It's a whitelist, sort of for security reasons...

Reply:
Okay.. let me see here. 
1. Map loops over all properties.
2. Maps argument is a arrow function
3. The arrow function deconstructs the map item
4. The arrow function returns the value it is assigned. 

And because of all these steps you end up with some filtered properties filtered out. 
I cant understand this from the code here.

Reply:
Not "some", but "all" except the whitelist. It's not too important because GraphQL only exposes properties which have a GraphQL schema definition, but the resolver functions are actually not dependant upon GraphQL and I'm sorta moving them over to lib-explorer and uses them as getters outside GraphQL. So I'm sorta making a uniform way of doing things, and basically yes, that is approaching typing. So I might go all the way into typescript eventually. Little by little.